src/swoop/db/migrations/00006_workflow_constraint.up.sql

```diff
@@ -0,0 +1,14 @@
+ALTER TABLE swoop.action
+  DROP CONSTRAINT workflow_or_callback
+, ADD CONSTRAINT workflow_or_callback CHECK (
+    CASE
+      WHEN action_type = 'callback'
+        THEN
+          parent_uuid IS NOT NULL
+          AND payload_uuid IS NULL
+      WHEN action_type = 'workflow' THEN
+        action_name IS NOT NULL
+        AND payload_uuid IS NOT NULL
+        AND parent_uuid IS NULL
```

Comment:
I don't think you can set this constraint unless you also null the parent_uuid for all rows first. In other words, I think this migration needs to start with:

```
UPDATE swoop.action SET parent_uuid = NULL where action_type = 'workflow';
```